app/src/org/commcare/utils/UriToFilePath.java

```diff
@@ -102,8 +110,12 @@ private static String getDataColumn(Context context, Uri uri, String selection,
             cursor = context.getContentResolver().query(uri, projection,
                     selection, selectionArgs, null);
             if (cursor != null && cursor.moveToFirst()) {
-                final int column_index = cursor.getColumnIndexOrThrow(column);
-                return cursor.getString(column_index);
+                try {
+                    final int column_index = cursor.getColumnIndexOrThrow(column);
+                    return cursor.getString(column_index);
+                } catch (IllegalArgumentException e) {
+                    throw new NoDataColumnForUriException();
```

Comment:
This method should declare that it throws this exception so that all usages are forced to handle it